lib/vanagon/platform/windows.rb

```diff
@@ -34,6 +34,143 @@ def package_name(project)
+      # @return [String] Wrapped Powershell Command
+      def install_chocolately_ps1(url)
+        batch_contents = <<-BATCH
+$ErrorActionPreference = 'Stop'
+
+try {
+  if ($env:TEMP -eq $null) {
+    $env:TEMP = Join-Path $env:SystemDrive 'temp'
+  }
+  $chocTempDir = Join-Path $env:TEMP "chocolatey"
+  $tempDir = Join-Path $chocTempDir "chocInstall"
+  if (![System.IO.Directory]::Exists($tempDir)) {[System.IO.Directory]::CreateDirectory($tempDir)}
+  $file = Join-Path $tempDir "chocolatey.zip"
+  $chocErrorLog = Join-Path $tempDir "chocError.log"
+
+  # download the package
+  $url = "#{url}" + 'Packages()?$filter=((Id%20eq%20%27chocolatey%27)%20and%20(not%20IsPrerelease))%20and%20IsLatestVersion'
+  $downloader = new-object System.Net.WebClient
+  $downloader.Proxy.Credentials=[System.Net.CredentialCache]::DefaultNetworkCredentials;
+
+  Write-Output "Querying latest package from $url"
+  [xml]$pkg = $downloader.DownloadString($url)
+  $url = $pkg.feed.entry.content.src
+
+  Write-Output "Downloading $url to $file"
+  $downloader.DownloadFile($url, $file)
+
+  # Unzip the package
+  Write-Output "Extracting $file to $tempDir..."
+  $shellApplication = new-object -com shell.application
+  $zipPackage = $shellApplication.NameSpace($file)
+  $destinationFolder = $shellApplication.NameSpace($tempDir)
+  $destinationFolder.CopyHere($zipPackage.Items(),0x10)
+
+  # call chocolatey install
+  Write-Output "Installing chocolatey on this machine"
+  $toolsFolder = Join-Path $tempDir "tools"
+  $chocInstallPS1 = Join-Path $toolsFolder "chocolateyInstall.ps1"
+
+  if ($PSVersionTable.psversion.Major -gt 2) {
+    & $chocInstallPS1
+  }
+  else {
+    $output = Invoke-Expression $chocInstallPS1
+    $output
+    write-output "Any errors that occured during install or upgrade are logged here: $chocoErrorLog"
+    $error | out-file $chocErrorLog
+  }
+
+  $chocInstallVariableName = "ChocolateyInstall"
+  $chocoPath = [Environment]::GetEnvironmentVariable($chocInstallVariableName, [System.EnvironmentVariableTarget]::User)
+  if ($chocoPath -eq $null -or $chocoPath -eq '') {
+    $chocoPath = 'C:\\ProgramData\\Chocolatey'
+  }
+
+  Write-Output 'Ensuring chocolatey.nupkg is in the lib folder'
+  $chocoPkgDir = Join-Path $chocoPath 'lib\\chocolatey'
+  $nupkg = Join-Path $chocoPkgDir 'chocolatey.nupkg'
+  if (![System.IO.Directory]::Exists($chocoPkgDir)) { [System.IO.Directory]::CreateDirectory($chocoPkgDir); }
+  Copy-Item "$file" "$nupkg" -Force -ErrorAction SilentlyContinue
+}
+catch
+{
+  Write-Host "$($_.Exception.Message)"
+  exit 1
+}
+        BATCH
+      end
+      # Enclose the supplied command with the Powershell wrapper.
+      #
+      # @param command [String] optional repo target for built packages defined
+      #   at the project level
+      # @return [String] Wrapped Powershell Command
+      def powershell_command(command)
+        result = "powershell.exe -NoProfile -ExecutionPolicy Unrestricted -InputFormat None -Command #{command}"
```

Comment:
You may want to do something similar to what we do when we look for the powershell.exe in known locations prior to settling on one, see https://github.com/puppetlabs/puppetlabs-powershell/blob/e896f8b4546b4c34d1144fb6fa31da2df78274c7/lib/puppet/provider/exec/powershell.rb#L6-L13